### DISCUSSION TITLE
WKT-String an das Profilservice wird in CalcCRS erstellt

### DISCUSSION THREAD

**Discussion by fanie-mak** [e-netze/webgis-community#16] [General]
Hallo liebe WebGISler, 
uns ist aufgefallen, dass bei Erstellung des WKT-Strings im Geoland-Viewer etwas passiert ist. Anscheinend kommt das CalcCRS (31287) zum Einsatz und nicht die Projektion des Karten-Extents (3857). Dieses Verhalten können wir auch in KAGIS Maps nachvollziehen. 

Wir haben das gelöst, indem wir in der custom.js die Projektion beim URL des Profilservices auf das CalcCRS nachgezogen haben. Der Aufruf sollte aber mit CRS 3857 funktionieren. 

Geoland-Viewer, Build 5.22.4503
Hat in Vorversionen noch mit der Karten-Extent Projektion funktioniert.

https://voibos.rechenraum.com/voibos/voibos?name=profilservice&beschriftung=Profilauswertung&ueberhoehung=1&hintergrund=bmapgrau&stuetzpunktabstand=1&polygonzug=LINESTRING(421057.7231808607%20340585.1192557458,422221.3378530564%20340260.5636623511)&crs=31287&output=Profil,Lage,Tabelle 

LG

**Comment by gis-eni** [comment]
Es soll auch so sein, das für diese Dinge immer der CalcCRS verwendet wird. Anscheinend ist ein anderer "Bug" behoben worden, der jetzt auch diese Funktion richtig gestellt hat.

Es ist grundsätzlich immer schwierig wenn das Kartenkoordinatensystem nicht mit dem Calc-Koordinatensystem zusammen passt. 
Zur Info: Das Calc-Koordinatensystem ist das, in dem "gerechnet" wird. Wenn also beispielsweise ein Fläche oder eine Abstand mit dem Messwerkzeug ermittelt wird, geschieht das im Calc-Koordinatensystem. Alle Sketchtools versuchen die Koordinaten standardmäßig in diesem System zu verwenden. Das betrifft alles mögliche wie die Konstruktionstools (Bogenschnitt, ...) oder nächste Punkt per Abstand und Entfernung...
Darum wirkt sich das auch auf den Sketch für euer benutzerdefiniertes Werkzeug aus!

Man kann das Calc-Koordiantensystem in der ``cusotm.js`` global oder für eine einzelne Karte setzen:

```javascript
calcCrs = 31256;

if (mapUrlName === "Austria") {
    calcCrs = webgis.calc.crs_BestAustria_GK;  // this a function!!
};
```

Wenn man ganz genau sein will, kann man als Calc-Koordinatensystem auch eine Funktion schreiben die aufgrund der aktuellen Koordinaten das System bestimmt. Hier im Beispiel hab ich die vordefinierte Funktion ``webgis.calc.crs_BestAustria_GK`` angegeben: Je nachdem, wo man sich befindet wird hier entweder GK-M28, GK-M31 oder GK-M34 zum Rechnen verwendet.

Wenn man ``calcCrs`` als Funktion definiert, hat man natürlich im oben beschriebenen Fall das Problem, dass sich der Wert für den Parameter ``srs`` ändert, je nachdem wo man sich befindet.

Darum empfehle ich, die den Sketch immer im WGS84 zu übergeben. In der Url beim benutzerdefinierten Werkzeug erreicht man das mit:
``...&polygonzug={wkt-4326}&crs=4326``   (den Parameter gibt es nur für 4326!!!)

**Comment by gis-eni** [comment]
PS:
Als Erweiterung wird es im nächsten Build für diese benutzerdefinierten Werkzeug noch die Platzhalter ``{calc-srs}`` und {sketch-srs}`` geben.
Beides sollte in der Regel die gleiche Werte liefern. Welcher immer der Richtige sein wird, werden die Anwendungsbeispiele zeigen.
Den Link für das Benutzerdefinierte Werkzeug muss man dann so umbauen:

``...&polygonzug={wkt}&crs={calc-srs}``

``...&polygonzug={wkt}&crs={sketch-srs}``

``...&polygonzug={wkt-4326}&crs=4326``
